exp-player/addon/randomizers/pref-phys.js

```diff
@@ -10,105 +12,1336 @@ function shuffleArray(array) {
+        ['cup', 'whiteball', 'lotion', 'spray', 'apple', 'orangeball'],
+        ['orangeball', 'cup', 'lotion', 'spray', 'apple', 'whiteball'],
+        ['apple', 'whiteball', 'lotion', 'cup', 'spray', 'orangeball'],
+        ['whiteball', 'apple', 'orangeball', 'spray', 'cup', 'lotion'],
+        ['cup', 'whiteball', 'orangeball', 'spray', 'apple', 'lotion'],
+        ['orangeball', 'lotion', 'apple', 'whiteball', 'cup', 'spray'],
+        ['whiteball', 'orangeball', 'lotion', 'spray', 'apple', 'cup'],
+        ['whiteball', 'cup', 'lotion', 'spray', 'apple', 'orangeball'],
+        ['apple', 'cup', 'orangeball', 'whiteball', 'lotion', 'spray'],
+        ['orangeball', 'lotion', 'apple', 'cup', 'spray', 'whiteball'],
+        ['apple', 'lotion', 'cup', 'spray', 'whiteball', 'orangeball'],
+        ['cup', 'orangeball', 'lotion', 'spray', 'apple', 'whiteball'],
+        ['whiteball', 'orangeball', 'cup', 'apple', 'spray', 'lotion'],
+        ['cup', 'lotion', 'whiteball', 'spray', 'apple', 'orangeball'],
+        ['cup', 'orangeball', 'lotion', 'spray', 'whiteball', 'apple'],
+        ['lotion', 'cup', 'orangeball', 'spray', 'whiteball', 'apple'],
+        ['lotion', 'orangeball', 'cup', 'whiteball', 'apple', 'spray'],
+        ['apple', 'whiteball', 'cup', 'lotion', 'spray', 'orangeball'],
+        ['cup', 'lotion', 'apple', 'spray', 'whiteball', 'orangeball'],
+        ['apple', 'lotion', 'orangeball', 'spray', 'whiteball', 'cup'],
+        ['whiteball', 'cup', 'orangeball', 'apple', 'spray', 'lotion'],
+        ['whiteball', 'apple', 'orangeball', 'spray', 'lotion', 'cup'],
+        ['whiteball', 'cup', 'lotion', 'apple', 'spray', 'orangeball'],
+        ['apple', 'orangeball', 'cup', 'spray', 'whiteball', 'lotion'],
+        ['whiteball', 'apple', 'orangeball', 'cup', 'lotion', 'spray'],
+        ['apple', 'whiteball', 'orangeball', 'spray', 'cup', 'lotion'],
+        ['cup', 'apple', 'orangeball', 'lotion', 'whiteball', 'spray'],
+        ['cup', 'orangeball', 'apple', 'lotion', 'whiteball', 'spray'],
+        ['whiteball', 'cup', 'orangeball', 'lotion', 'spray', 'apple'],
+        ['orangeball', 'whiteball', 'apple', 'cup', 'spray', 'lotion'],
+        ['whiteball', 'orangeball', 'apple', 'lotion', 'spray', 'cup'],
+        ['cup', 'orangeball', 'whiteball', 'apple', 'spray', 'lotion'],
+        ['apple', 'whiteball', 'lotion', 'spray', 'cup', 'orangeball'],
+        ['orangeball', 'cup', 'whiteball', 'spray', 'apple', 'lotion'],
+        ['orangeball', 'apple', 'cup', 'lotion', 'whiteball', 'spray'],
+        ['cup', 'apple', 'lotion', 'spray', 'whiteball', 'orangeball'],
+        ['orangeball', 'apple', 'whiteball', 'spray', 'cup', 'lotion'],
+        ['cup', 'orangeball', 'whiteball', 'lotion', 'spray', 'apple'],
+        ['orangeball', 'whiteball', 'apple', 'cup', 'lotion', 'spray'],
+        ['whiteball', 'lotion', 'orangeball', 'apple', 'spray', 'cup'],
+        ['whiteball', 'lotion', 'orangeball', 'apple', 'cup', 'spray']
```

Comment:
Ah, so this is a bit tricky. These lists are needed for counterbalancing to distribute object-comparison mappings *roughly* evenly across kids, and then to ensure (without requiring complicated queries about their individual histories) that those kids see different pairings as they keep participating. Cycling through a list of such mappings, in a fixed pseudorandom order, is one of the simpler and more transparent/reproducible ways to arrange this. 

It's critical here that the lists are in a **fixed** but (pseudo-)random order; they're of course programmatically generated, but only once. It would be possible to move the code that generates them to the randomizer, but we'd need to seed whatever random generator was used and be absolutely certain that changes to any underlying code could never affect the lists generated. (And yes, not all of them actually list all the permutations--gravity needs to leave out ones that pair 'table' with 'spray' or 'ramp' with 'orangeball', inertia and control have some redundancy removed because we're repeating the same comparison types twice each.) My best guess was that loading this bunch of text was worth it for better scientific transparency, since we're talking about a file size orders of magnitude less than the videos--does that sound reasonable?

Another option is to basically manually compress this data--there's of course no reason that these permutations have to explicitly list the strings over and over; we could represent each permutation with a number and then parse those out. But that would actually make it harder to see what's going on.

A third option would be to subsample from the list of possible permutations. However, scientifically it's better to be sampling from the whole distribution, which will necessarily reduce the variance in e.g. how many kids see 'whiteball' paired with 'stop hand/nohand' more often than with 'reverse barrier/nobarrier'. 

So basically yes, this is terrible, it can be moved to another file if that would be nicer to maintain, but we do have reasons for doing it.

* The list is in a **fixed**, but (pseudo-)random, order. (It defeats the purpose for the list to be shuffled upon 
*